appendix-basic65.tex

```diff
@@ -220,168 +252,240 @@ \subsection{BASIC 65 Variables}
+variables with longer names. It otherwise behaves like any other variable.
+
 \subsection{BASIC 65 Arrays}
 \index{BASIC 65 Arrays}
 
-Each array consumes the number of elements multiplied by the item size,
-plus the size of the header (6 + 2 * dimensions) in memory.
-For example the array
-\begin{tcolorbox}[colback=black,coltext=white]
-\verbatimfont{\codefont}
-\begin{verbatim}
-100 DIM X(8,2,3)
-\end{verbatim}
-\end{tcolorbox}
-has 3 dimensions and 108 (9 x 3 x 4) items. You might be asking: Why is it 9 x 3 x 4,
-when the program uses 8, 2, and 3? This is because array indexes start at 0, not 1.
-
-The size for real items is 5, so the data of that array above would occupy 540 (5 x 108) bytes.
-The header size is 12 bytes (6 + 2 * 3), so the total length in memory is 552 bytes (540 + 12).
-
-\begin{comment}
-Arrays are stored in bank 1 starting at address \$2000 and expand upwards.
-They share the available memory at \$2000 .. \$F6FF with the string area,
-which starts in bank 1 at address \$F6FF, and expands \textit{downwards}.
-\end{comment}
-Each of the above scalar variable types can be used as an array, by declaring
-them with a {\bf DIM} statement. The arrays are initialised to zero for all
-elements on declaration. If an undeclared array element is used,
-an automatic implicit declaration is performed, which sets the upper boundary
-for each dimension to 10. For example, the usage of an undeclared element
-\screentext{AB(3,5)} would automatically perform a \screentext{DIM AB(10,10)}.
-As noted previously, the lower boundary for each dimension is always 0 (zero),
-so an array initialised with \screentext{DIM AB(10)} consists of 11 elements and accepts indexes from
-0 to 10.
-
-String arrays are more precisely expressed as \textit{arrays of string
-descriptors}. Each item consists of three bytes, which hold
-these values: The length of the string, and the 2 byte address (low/high byte)
-of the string in string memory.
-The usage of the BASIC function {\bf POINTER} with a string or
-string array element as the argument, returns the address of the descriptor, not the string itself.
-
-\setlength{\tabcolsep}{1mm}
-\begin{center}
-\begin{tabular}{|l|l|l|l|l|}
-\hline
-\multicolumn{2}{|c|}{\bf Type \& Item Size}  & {\bf Appended}    & {\bf Range}         & {\bf Example}  \\
-\multicolumn{2}{|c|}{}                      & {\bf Character}   &                     &                \\
-\hline
-Byte     Array &  1                         &    \screentext{\&} & 0 .. 255            & \screentext{BY\&(5,6) = 23}   \\
-Integer  Array &  2                         &    \screentext{\%} & -32768 .. 32767     & \screentext{I\%(0,10) = 5}    \\
-Real     Array &  5                         &    none            & -1E37 .. 1E37       & \screentext{XY(I\%) = 1/3}    \\
-String   Array &  3                         &    \screentext{\$} & 0 .. 255 characters & \screentext{AB\$(X) = "TEXT"} \\
-\hline
-\end{tabular}
-\end{center}
```

Comment:
For the "Arrays" section, re: 2nd table of Byte/Integer/Real/String that got removed

Yeah, I don't mind the deletion of this 2nd table, it was very similar to the 1st.

Though it did have that extra 'size' column to show the number of bytes each entry in the array consumed.

Could such detail be added to the earlier table instead?

Reply:
The POINTER entry is the best place for this for now. I have added memory representation notes to that entry.